synapse/storage/databases/main/room.py

```diff
@@ -699,20 +699,30 @@ def delete_ratelimit_txn(txn: LoggingTransaction) -> None:
         await self.db_pool.runInteraction("delete_ratelimit", delete_ratelimit_txn)
 
     @cached()
-    async def get_retention_policy_for_room(self, room_id: str) -> Dict[str, int]:
+    async def get_retention_policy_for_room(
+        self, room_id: str
+    ) -> Dict[str, Optional[int]]:
         """Get the retention policy for a given room.
 
         If no retention policy has been found for this room, returns a policy defined
         by the configured default policy (which has None as both the 'min_lifetime' and
         the 'max_lifetime' if no default policy has been defined in the server's
         configuration).
 
+        If support for retention policies is disabled, a policy without any bound is
+        returned.
```

Comment:
I'm not sure what to expect for "a policy without any bound", maybe the following is clearer:

```suggestion
        If support for retention policies is disabled, a policy with a 'min_lifetime' and 'max_lifetime'
        of None is returned.
```